### PR TITLE
Fix: Klog Flush during entry and exit.

### PR DESCRIPTION
**Pull request by Ealianis**
### Description of your changes
Updated kLog and added flushing to both the entry and exit of the module.

I have:
- [x] Read and followed Caravel's [Code of conduct](https://github.com/Azure/k8s-work-api/blob/master/code-of-conduct.md).
- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested
Ran tests locally, build and deployed image to local kind cluster.

### CODE REVIEW

File: cmd/workcontroller/workcontroller.go [hunk 1]
```diff
@@ -49,6 +47,10 @@ func init() {
 	utilruntime.Must(v1alpha1.AddToScheme(scheme))
 }
 
+func init() {
+	klog.InitFlags(nil)
+}
```
Comment (ryanzhang-oss) [inline cmd/workcontroller/workcontroller.go:52]: why is this moved to init?

Comment (Ealianis) [reply cmd/workcontroller/workcontroller.go:52]: To ensure the the command-line flags, parsed from os.Args are set prior to anything in the main func.

Comment (Ealianis) [reply cmd/workcontroller/workcontroller.go]: BTW I combined it to the single Init. 
https://github.com/Azure/k8s-work-api/pull/39/files#diff-1a0fb6ecc34795cbd8382df9c61b44dc454034081c3381041a0f113ec96ee457R48

Comment (ryanzhang-oss) [reply cmd/workcontroller/workcontroller.go]: I am not sure if I asked the right question.  Let me try again, I wonder why does klog's flag need to set prior to anything in the main func?

File: cmd/workcontroller/workcontroller.go [hunk 2]
```diff
@@ -99,12 +97,12 @@ func main() {
 	}
 	if err != nil {
 		setupLog.Error(err, "error reading kubeconfig to connect to hub")
-		os.Exit(1)
+		klog.FlushAndExit(klog.ExitFlushTimeout, 1)
```
Comment (ryanzhang-oss) [inline cmd/workcontroller/workcontroller.go:100]: I wonder if there is any reason to flush here since there is already a deferred klog.Flush() ? Will that flush fail?

File: cmd/workcontroller/workcontroller.go [hunk 3]
```diff
@@ -60,6 +58,7 @@ func main() {
 	var healthAddr string
 	var concurrentReconciles int
 
+	klog.InitFlags(nil)
```
Comment (ryanzhang-oss) [inline cmd/workcontroller/workcontroller.go:61]: I wonder if there is any reason to put this ahead of the rest of the flag?

Comment (Ealianis) [reply cmd/workcontroller/workcontroller.go:61]: @ryanzhang-oss, init Flag does two important things prior to our flag setting. It first constructs the FlagSet struct, setting the FlagSet's name, error handling, and the usage print message format. It also set's kLogs default flags.  Without the init first, there would not be struct to set the new flags within.

For a moment I also thought that we wouldn't want to override Flag's default values / usage strings, but it turns out that is protected by a panic if there is a conflict on the names.

File: cmd/workcontroller/workcontroller.go (hunk 2 above)
Comment (Ealianis) [reply cmd/workcontroller/workcontroller.go:100]: @ryanzhang-oss ,
If you call os.Exit(#) the deferred functions are not run.
https://github.com/golang/go/blob/master/src/os/proc.go#L59

Comment (Ealianis) [reply cmd/workcontroller/workcontroller.go:100]: As for if there is an issue, the mutex is locked. The secondary gorotuine will be blocked until the mutex is unlocked. The Flush & Sync errors (if any) are ignored within.

**Review by aagusuab — APPROVED** [review]
LGTM
